response_operations_ui/controllers/party_controller.py

```diff
@@ -0,0 +1,75 @@
+import logging
+
+import requests
+from structlog import wrap_logger
+
+from response_operations_ui import app
+from response_operations_ui.exceptions.exceptions import ApiError
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def get_party_by_ru_ref(ru_ref):
+    logger.debug('Retrieving reporting unit', ru_ref=ru_ref)
+    url = f'{app.config["PARTY_URL"]}/party-api/v1/parties/type/B/ref/{ru_ref}'
+    response = requests.get(url, auth=app.config['PARTY_AUTH'])
+
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        logger.error('Failed to retrieve reporting unit', ru_ref=ru_ref)
```

Comment:
The get party api can respond with a 400 or 404 which we don't want to log at error level.

```
log_level = logger.warning if response.status_code in (400, 404) else logger.exception
log_level("Respondent retrieval failed", ru_ref=ru_ref)
```